### ISSUE TITLE
[HOLD for payment 2024-04-09] [$500] Workspace - New Workspace button doesn't create WS and shows no default name

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#38420]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.53-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exploratory around https://expensify.testrail.io/index.php?/tests/view/4428235
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to NewDot and login
2. Click WS dropdown in the top of LHN
3. Click +
4. Change WS name and click Save
5. Click New Workspace button in the top right corner

## Expected Result:
New WS created, default name should be displayed

## Actual Result:
Default name is not displayed, and WS is not created

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/f43e8a9a-90bf-47d0-9409-0ddee4c0e068

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0171b0f044271b7cff</li>
        <li>Upwork Job ID: 1769593725439668224</li>
        <li>Last Price Increase: 2024-03-18</li>
<li>Automatic offers: </li>
<ul>
<li>alitoshmatov | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @alitoshmatov</details>

**Comment by lanitochka17** [comment]
@kadiealexander  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by iammudassirali** [comment]
Hi there!

I've carefully reviewed the issue you've reported regarding the default workspace creation problem in the Expensify app. I understand the importance of maintaining a reliable and seamless user experience, especially when it comes to financial transactions.

To address this issue, I propose the following technical solution:

**Investigation and Analysis:**

I'll start by examining the codebase related to workspace creation in the Expensify app.
I'll identify any potential logic errors or misconfigurations that could be causing the default workspace name not to display.
**Code Refactoring:**

If necessary, I'll refactor the relevant code to ensure that the default workspace name is properly displayed.
This may involve debugging the existing code and making modifications to improve its functionality.
**Testing and Validation:**

Once the changes are implemented, I'll thoroughly test the workspace creation functionality across different environments, including staging and production.
I'll also perform regression testing to ensure that no new issues are introduced as a result of the changes.
**Documentation and Reporting:**

Throughout the process, I'll maintain detailed documentation of the steps taken and the changes made.
I'll provide clear reports on the findings, actions taken, and outcomes of the testing phase.
**Collaboration and Communication:**

I'll maintain open communication with the Expensify team, providing regular updates on the progress of the resolution process.
I'll actively seek feedback and collaborate with team members to ensure that the solution aligns with the project requirements.
I'm confident that my technical expertise and problem-solving skills make me well-suited to address this issue effectively. I'm committed to delivering a high-quality solution that meets the standards of reliability and user experience expected from the Expensify app.

I look forward to the opportunity to contribute to the success of the Expensify project. If you have any questions or need further clarification, please don't hesitate to reach out.

Best regards,
Mudassir Ali

P.S. I'm excited about the opportunity to work with the Expensify team and contribute to the ongoing improvement of the app! 😊

**Comment by iammudassirali** [comment]
Contributor details
Email Address: thisismudassirali@gmail.com
Upwork Profile: https://www.upwork.com/freelancers/~0191fe0bea01c9f099

**Comment by iammudassirali** [comment]
Contributor details
Your Expensify account email: thisismudassirali@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0191fe0bea01c9f099

**Comment by iammudassirali** [comment]
Hi there!

I've carefully reviewed the issue you've reported regarding the default workspace creation problem in the Expensify app. I understand the importance of maintaining a reliable and seamless user experience, especially when it comes to financial transactions.

To address this issue, I propose the following technical solution:

Investigation and Analysis:

I'll start by examining the codebase related to workspace creation in the Expensify app.
I'll identify any potential logic errors or misconfigurations that could be causing the default workspace name not to display.
Code Refactoring:

If necessary, I'll refactor the relevant code to ensure that the default workspace name is properly displayed.
This may involve debugging the existing code and making modifications to improve its functionality.
Testing and Validation:

Once the changes are implemented, I'll thoroughly test the workspace creation functionality across different environments, including staging and production.
I'll also perform regression testing to ensure that no new issues are introduced as a result of the changes.
Documentation and Reporting:

Throughout the process, I'll maintain detailed documentation of the steps taken and the changes made.
I'll provide clear reports on the findings, actions taken, and outcomes of the testing phase.
Collaboration and Communication:

I'll maintain open communication with the Expensify team, providing regular updates on the progress of the resolution process.
I'll actively seek feedback and collaborate with team members to ensure that the solution aligns with the project requirements.
I'm confident that my technical expertise and problem-solving skills make me well-suited to address this issue effectively. I'm committed to delivering a high-quality solution that meets the standards of reliability and user experience expected from the Expensify app.

I look forward to the opportunity to contribute to the success of the Expensify project. If you have any questions or need further clarification, please don't hesitate to reach out.

Best regards,
Mudassir Ali

P.S. I'm excited about the opportunity to work with the Expensify team and contribute to the ongoing improvement of the app! 😊

**Comment by gieoon** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
1. The default workspace name is not defined when creating a workspace from `WorkspaceSwitcherPage.tsx`. 
1. The workspace is not created or saved.

### What is the root cause of that problem?

Workspace is being created from a policy draft rather than a policy. The current logic does not handle policy drafts and the name is undefined in `WorkspaceProfilePage`.

WorkspaceProfilePage does not accept policyDraft, so policy is undefined and the policyDraft is ignored. 
https://github.com/Expensify/App/blob/e737a63615e600e5a76a4268d288389318d747ce/src/pages/workspace/WorkspaceProfilePage.tsx#L44

The policy draft is also never saved, so `App.savePolicyDraftByNewWorkspace(policyDraft.id, policyDraft.name, '', policyDraft.makeMeAdmin);` needs to be called from somewhere. 

### What changes do you think we should make in order to solve the problem?

Policy needs to be inherited by `WorkspaceProfilePage` and handled in the logic when creating a policy.
```
function WorkspaceProfilePage({policyDraft, policy, currencyList = {}, route}: WorkSpaceProfilePageProps) {
      
      policy = policyDraft?.id ? policyDraft : policy;

}
```

`App.savePolicyDraftByNewWorkspace` also needs to be called from within `useEffect` in `WorkspaceProfilePage.tsx` as it is also done in `WorkspaceInitialPage.tsx`.
https://github.com/Expensify/App/blob/e737a63615e600e5a76a4268d288389318d747ce/src/pages/workspace/WorkspaceInitialPage.tsx#L76-L86

Proposed solution in `WorkspaceProfilePage.tsx`
```
useEffect(() => {
        
        ...

        if ( policyDraft ) {
            App.savePolicyDraftByNewWorkspace(policyDraft.id, policyDraft.name, '', policyDraft.makeMeAdmin);
        }

        ...

    }, []);
```

### What alternative solutions did you explore? (Optional)
Changing the props directly in the HOC `withPolicy.tsx` before they reach the React Component.

**Comment by gieoon** [comment]
Contributor details
Your Expensify account email: jun.a.kagaya@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01506cc9754f8fc72b

**Comment by iammudassirali** [comment]
**Please re-state the problem that we are trying to solve in this issue.**
The issue at hand involves the failure to display the default workspace name when creating a new workspace in the Expensify app. This results in user confusion and impacts the overall user experience, especially for first-time users.

**What is the root cause of that problem?**
Upon analysis, it appears that there is a logic error or misconfiguration in the code related to workspace creation. This prevents the default workspace name from being properly retrieved and displayed to the user during the creation process.

**What changes do you think we should make in order to solve the problem?**
To address this issue, we need to identify and rectify the underlying cause within the codebase. This may involve debugging the existing code and implementing modifications to ensure that the default workspace name is correctly retrieved and displayed to the user.

**What alternative solutions did you explore? (Optional)**
While exploring alternative solutions, we considered the possibility of implementing a workaround to manually input the default workspace name. However, this approach would not address the root cause of the issue and could lead to further user confusion.

In summary, the proposed solution involves identifying and rectifying the logic error or misconfiguration within the codebase to ensure that the default workspace name is correctly displayed during the creation process. This will contribute to a more seamless and user-friendly experience for Expensify app users.

Best regards,
**Mudassir Ali**

**Comment by kaushiktd** [comment]
### Please re-state the problem that we are trying to solve in this issue.
 New Workspace button doesn't create WS and shows no default name

### What is the root cause of that problem?
When attempting to create a new workspace after already creating one, the policy object returns undefined, causing issues with the application logic.

### What changes do you think we should make in order to solve the problem?

The policy object received as props is undefined in certain scenarios, particularly during workspace creation. To mitigate this issue, we obtain the policy ID from the route parameters and utilize it to fetch the corresponding policy object. This ensures that the policy is correctly retrieved and resolves the undefined policy error.

add here
[](https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceProfilePage.tsx#L64)
```
  React.useEffect(() => {
        const policyDraftId = route.params?.policyID;

        if (!policyDraftId) {
            return;
        }

        App.savePolicyDraftByNewWorkspace(policyDraftId);

    }, []);
```

video:-https://drive.google.com/file/d/1VkKQacoakwwl9X2And_f74MjNEu0Ltib/view?usp=sharing

**Comment by WojtekBoman** [comment]
There are two ways to fix this, we have to decide which option is better for us. This issue is strongly related to the flow of opening the workspace settings, and we need to determine which flow is correct:

1.  When we are redirected to WorkspaceInitialScreen from the deeplink (for example `https://dev.new.expensify.com:8082/settings/workspaces/:id`) and then click the back button, we'll be navigated to **the workspace list**. In this flow when user creates a new workspace from the switcher (chat tab), and then goes back to the previous screen, **the workspace list** will be displayed (profile tab).


https://github.com/Expensify/App/assets/47774969/a0ac81ad-d80a-4243-9d09-97c7bbc80721



2. When we are redirected to WorkspaceInitialScreen from the deeplink and then click the back button, we'll be navigated to **the home screen**. In this flow when user creates a new workspace from the switcher, and then goes back to the previous screen, **the home page** will be displayed. 


https://github.com/Expensify/App/assets/47774969/7233acfc-8479-4b0b-97ac-f9e3ddd41dea





I need to know how the app should behave in this case and then I'll be able to fix it :)

cc: @mountiny @hayata-suenaga @adamgrzybowski

**Comment by mountiny** [comment]
@WojtekBoman I believe the first option is better cc @trjExpensify

**Comment by trjExpensify** [comment]
Can't we keep the bottom tab position they were on when they opened the switcher? Is that out of the question for some reason?

**Comment by trjExpensify** [comment]
Chatted to Vit and it seems like (1) is most consistent with our UP navigation, so let's start there. 👍

**Comment by kaushiktd** [comment]
@WojtekBoman Do you have any feedback for my [proposal](https://github.com/Expensify/App/issues/38420#issuecomment-2007003275)?

**Comment by mountiny** [comment]
$500 to @alitoshmatov

**Comment by kadiealexander** [comment]
@alitoshmatov please don't forget about the checklist!

**Comment by kadiealexander** [comment]
Bumped @alitoshmatov [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1712812944180559).

**Comment by alitoshmatov** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/33280
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/33280/files#r1561250597
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: No need
- Determine if we should create a regression test for this bug. No need

**Comment by kadiealexander** [comment]
Payouts due:

- [x] Contributor+: $500 @alitoshmatov 

Upwork job is [here](https://www.upwork.com/ab/feedbacks/c/1328455843768528896/contract/36739201).
